vertical-pod-autoscaler/enhancements/8818-in-place-only/README.md

```diff
@@ -0,0 +1,256 @@
+In these scenarios, users would prefer:
+
+- To operate with current (potentially suboptimal) resource allocations until an in-place update becomes feasible
+- To receive clear signals when updates cannot be applied
+- To have VPA continuously retry updates as cluster conditions change
+
+## Goals
+
+- Provide a truly non-disruptive VPA update mode that never evicts pods
+- Allow VPA to eventually apply updates when cluster conditions improve
+- Respect the existing in-place update infrastructure from AEP-4016
+
+## Non-Goals
+
+- Guarantee that all updates will eventually succeed (node capacity constraints may prevent this)
+- Provide mechanisms to automatically increase node capacity to accommodate updates
+- Change the behavior of existing update modes (Off, Recreate, InPlaceOrRecreate)
+- Eliminate all possible disruption scenarios (see [Risk Mitigation](#risk-mitigation) for details on memory limit downsizing risks)
+
+
+## Proposal
+
+Add a new supported value of UpdateMode: `InPlace`
+This mode will:
+- Apply recommendations during pod admission (like all other modes)
+- Attempt in-place updates for running pods under the same conditions as `InPlaceOrRecreate`
+- Never add pods to `podsForEviction` if in-place updates fail
+- Continuously retry failed in-place update
+
+## Design Details
+
+Add `UpdateModeInPlace` to the VPA types:
+
+```golang
+// In pkg/apis/autoscaling.k8s.io/v1/types.go
+const (
+    // ... existing modes ...
+    // UpdateModeInPlace means that VPA will only attempt to update pods in-place
+    // and will never evict them. If in-place update fails, VPA will rely on
+    // Kubelet's automatic retry mechanism.
+    UpdateModeInPlace UpdateMode = "InPlace"
+)
+```
+
+### Resize Status Handling
+
+The `InPlace` mode handles different resize statuses with distinct behaviors. Critically, before checking the resize status, VPA first compares the current recommendation against the pod's spec.resources. If they differ, VPA attempts to apply the new recommendation regardless of the current resize status, as the new recommendation may be feasible (e.g., a smaller resource request that fits on the node).
+
+- `Deferred`: When the resize status is Deferred and the recommendation matches spec, VPA waits and lets kubelet handle it. This means kubelet is waiting to apply the resize, and VPA should not interfere.
+- `Infeasible`: When the resize status is Infeasible and the recommendation matches spec, VPA defers action. The node cannot accommodate the current resize, but if the recommendation changes, VPA will attempt the new resize.
+- `InProgress`: When the resize status is InProgress and the recommendation matches spec, VPA waits for completion. The resize is actively being applied by kubelet.
+- `Error`: When the resize status is Error, VPA retries the operation. An error occurred during resize and retrying may succeed.
```

Comment:
What is meant by `Error` here? There are two interpretations I can think of:
- The API server returns an error (the patch does not succeed).
- A resize has the `InProgress` status, but there is an `Error` present in the `InProgress` condition. 

I assume you are talking about the first one, because "retrying" doesn't make sense with the second one (assuming recommendation matches spec). But I think both should be explicitly mentioned for clarity.